src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -482,35 +526,35 @@ class PostgresEntityDataQueryService(
     }
 
     private fun extractValues(propertyValues: Map<UUID, Set<Map<ByteBuffer, Any>>>): Map<UUID, Set<Any>> {
-        return propertyValues.mapValues { (entityKeyId, replacements) -> replacements.flatMap { it.values }.toSet() }
+        return propertyValues.mapValues { (_, replacements) -> replacements.flatMap { it.values }.toSet() }
     }
 
     /**
      * Tombstones all entities in an entity set.
      */
     private fun tombstone(conn: Connection, entitySetId: UUID): WriteEvent {
         check(!conn.autoCommit) { "Connection auto-commit must be disabled" }
-        val tombstoneVersion = -System.currentTimeMillis()
+        val version = System.currentTimeMillis()
 
 
         val numUpdated = conn.prepareStatement(updateVersionsForEntitySet).use { ps ->
-            ps.setLong(1, tombstoneVersion)
-            ps.setLong(2, tombstoneVersion)
-            ps.setLong(3, tombstoneVersion)
+            ps.setLong(1, -version)
+            ps.setLong(2, -version)
+            ps.setLong(3, -version)
             ps.setObject(4, entitySetId)
             ps.executeUpdate()
         }
 
         //We don't count property type updates.
         conn.prepareStatement(updateVersionsForPropertiesInEntitySet).use { ps ->
-            ps.setLong(1, tombstoneVersion)
-            ps.setLong(2, tombstoneVersion)
-            ps.setLong(3, tombstoneVersion)
+            ps.setLong(1, -version)
+            ps.setLong(2, -version)
+            ps.setLong(3, -version)
             ps.setObject(4, entitySetId)
             ps.executeUpdate()
         }
 
-        return WriteEvent(tombstoneVersion, numUpdated)
+        return WriteEvent(-version, numUpdated)
```

Comment:
I think the WriteEvent should return the nonnegative version -- these events are used for auditing, and the version for auditing should effectively just be the timestamp (so that when you query for events in a given time range, the tombstone events will also show up). I realize we're using negative versions for the WriteEvents in all the tombstone calls, but I think that's a bug

Reply:
Alternately we could force AuditingComponent to take the absolute value of the version when converting it to an OffsetDateTime

Reply:
Okay, that wasn't really clear to me, but makes sense. Will do both!